cinema/views.py

```diff
@@ -56,3 +76,58 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        movie_id = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie_id:
+            movie_id = int(movie_id)
+            queryset = queryset.filter(movie_id=movie_id)
+        if date:
+            date = date.split("-")
+            queryset = queryset.filter(
+                show_time__year=int(date[0]),
+                show_time__month=int(date[1]),
+                show_time__day=int(date[2]),
```

Comment:
Consider adding error handling for the date parsing logic to ensure that invalid date formats do not cause the application to crash. You could use a try-except block to catch any exceptions and handle them appropriately.